src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxIndexWriteOperation.cs

```diff
@@ -171,9 +171,12 @@ public override void Delete(LazyStringValue key, IndexingStatsScope stats)
                 _indexWriter.TryDeleteEntry(key.AsReadOnlySpan());
         }
 
-        public override void DeleteTimeSeries(LazyStringValue docId, LazyStringValue key, IndexingStatsScope stats)
+        public override void DeleteByPrefix(LazyStringValue key, IndexingStatsScope stats)
         {
-            // Lucene method
+            EnsureValidStats(stats);
+
+            using (Stats.DeleteStats.Start())
+                _indexWriter.DeleteByPrefix(key.ToString(CultureInfo.InvariantCulture));
```

Comment:
Why do we need to materialize `key` as `string`? Couldn't we work with `LazyStringValue`? As I see we'll convert it to `Slice` via `Slice.From()` but we do have an overload there which accepts `LazyStringValue`

Reply:
I see `LazyStringValue` has `AsReadOnlySpan()` method, we can pass it as well, without extra materialization of the key.

Reply:
done